src/View/Helper/UrlHelper.php

```diff
@@ -153,15 +153,12 @@ public function script($path, array $options = []): string
      *    Set to true to apply timestamps when debug is true. Set to 'force' to always
      *    enable timestamping regardless of debug value.
      *
-     * @param string|array $path Path string or URL array
+     * @param string $path Path string or URL array
      * @param array $options Options array.
      * @return string Generated URL
      */
-    public function assetUrl($path, array $options = []): string
+    public function assetUrl(string $path, array $options = []): string
     {
-        if (is_array($path)) {
-            return $this->build($path, $options);
-        }
```

Comment:
Is the amount of breakage that we're going to push onto users worth the small number of typehints? I think this will make template more verbose and complicated when the helpers should be helping make template code simple.

Reply:
With my deprecations and maybe some tooling fixes, yes.